I have tested these changes against Neptune and a timeout error now results in the following output:

```
Creating statement files

An error occurred while exporting RDF as Turtle. Elapsed time: 0 seconds


An error occurred while exporting rdf graph. Elapsed time: 1 seconds

java.lang.RuntimeException: X-Neptune-Status: 500 TimeLimitExceededException
X-Neptune-Detail: {"code":"TimeLimitExceededException","requestId":"dcc37392-6714-4e18-e301-efab45ec6ea2","detailedMessage":"Operation terminated (deadline exceeded)"}

        at com.amazonaws.services.neptune.rdf.NeptuneSparqlClient.executeTupleQuery(NeptuneSparqlClient.java:125)
        at com.amazonaws.services.neptune.rdf.ExportRdfGraphJob.lambda$execute$0(ExportRdfGraphJob.java:34)
        at com.amazonaws.services.neptune.util.Timer.timedActivity(Timer.java:41)
        at com.amazonaws.services.neptune.util.Timer.timedActivity(Timer.java:34)
        at com.amazonaws.services.neptune.rdf.ExportRdfGraphJob.execute(ExportRdfGraphJob.java:31)
        at com.amazonaws.services.neptune.ExportRdfGraph.lambda$run$0(ExportRdfGraph.java:77)
        at com.amazonaws.services.neptune.util.Timer.timedActivity(Timer.java:41)
        at com.amazonaws.services.neptune.util.Timer.timedActivity(Timer.java:34)
        at com.amazonaws.services.neptune.ExportRdfGraph.run(ExportRdfGraph.java:60)
        at com.amazonaws.services.neptune.export.NeptuneExportRunner.run(NeptuneExportRunner.java:72)
        at com.amazonaws.services.neptune.NeptuneExportCli.main(NeptuneExportCli.java:48)
Caused by: org.eclipse.rdf4j.query.QueryEvaluationException: Unkown record type: 123
        at org.eclipse.rdf4j.repository.sparql.query.SPARQLTupleQuery.evaluate(SPARQLTupleQuery.java:59)
        at com.amazonaws.services.neptune.rdf.NeptuneSparqlClient.executeTupleQuery(NeptuneSparqlClient.java:121)
        ... 10 more
Caused by: java.io.IOException: Unkown record type: 123
        at org.eclipse.rdf4j.query.resultio.binary.BinaryQueryResultParser.parse(BinaryQueryResultParser.java:188)
        at org.eclipse.rdf4j.query.resultio.AbstractTupleQueryResultParser.parseQueryResult(AbstractTupleQueryResultParser.java:48)
        at org.eclipse.rdf4j.http.client.SPARQLProtocolSession.getTupleQueryResult(SPARQLProtocolSession.java:699)
        at org.eclipse.rdf4j.http.client.SPARQLProtocolSession.sendTupleQuery(SPARQLProtocolSession.java:369)
        at org.eclipse.rdf4j.repository.sparql.query.SPARQLTupleQuery.evaluate(SPARQLTupleQuery.java:56)
        ... 11 more
An error occurred while exporting from Neptune: X-Neptune-Status: 500 TimeLimitExceededException
X-Neptune-Detail: {"code":"TimeLimitExceededException","requestId":"dcc37392-6714-4e18-e301-efab45ec6ea2","detailedMessage":"Operation terminated (deadline exceeded)"}
```

--- a/src/main/java/com/amazonaws/services/neptune/rdf/io/NeptuneExportSparqlRepository.java
+++ b/src/main/java/com/amazonaws/services/neptune/rdf/io/NeptuneExportSparqlRepository.java
@@ -58,18 +58,6 @@ public class NeptuneExportSparqlRepository extends SPARQLRepository {
         this.regionName = regionName;
         this.initAuthenticatingHttpClient();
 
-        HttpClientBuilder clientBuilder = HttpClientBuilder.create().addInterceptorLast((HttpResponseInterceptor) (response, context) -> {
-            lastContext = context;
-            HttpEntity entity = response.getEntity();
-            if (entity != null) {
-                context.setAttribute("raw-response-inputstream", entity.getContent());
-            }
-        });
-
-        SharedHttpClientSessionManager clientSessionManager = new SharedHttpClientSessionManager();
-        clientSessionManager.setHttpClientBuilder(clientBuilder);
-        this.setHttpClientSessionManager(clientSessionManager);
-
         Map<String, String> additionalHeaders = new HashMap<>();
         additionalHeaders.put("te", "trailers"); //Asks Neptune to send trailing headers which may contain error messages
         this.setAdditionalHttpHeaders(additionalHeaders);
@@ -80,6 +68,14 @@ public class NeptuneExportSparqlRepository extends SPARQLRepository {
         HttpClientBuilder httpClientBuilder = config.useSsl() ?
                 HttpClientBuilders.getSSLTrustAllHttpClientBuilder() :
                 HttpClientBuilder.create();
+
+        httpClientBuilder.addInterceptorLast((HttpResponseInterceptor) (response, context) -> {
+            lastContext = context;
+            HttpEntity entity = response.getEntity();
+            if (entity != null) {
+                context.setAttribute("raw-response-inputstream", entity.getContent());
+            }
+        });
 
         if (config.useIamAuth()) {
             v4Signer = new NeptuneApacheHttpSigV4Signer(regionName, awsCredentialsProvider);

--- a/src/main/java/com/amazonaws/services/neptune/rdf/io/NeptuneExportSparqlRepository.java
+++ b/src/main/java/com/amazonaws/services/neptune/rdf/io/NeptuneExportSparqlRepository.java
@@ -28,7 +28,6 @@ import org.apache.http.conn.EofSensorInputStream;
 import org.apache.http.impl.client.HttpClientBuilder;
 import org.apache.http.impl.io.ChunkedInputStream;
 import org.apache.http.protocol.HttpContext;
-import org.eclipse.rdf4j.http.client.SharedHttpClientSessionManager;
 import org.eclipse.rdf4j.http.client.util.HttpClientBuilders;
 import org.eclipse.rdf4j.repository.sparql.SPARQLRepository;
 
@@ -128,15 +127,16 @@ public class NeptuneExportSparqlRepository extends SPARQLRepository {
             return "";
         }
         Header[] trailers = chunkedInStream.getFooters();
-        String message = "";
+        StringBuilder messageBuilder = new StringBuilder();
         for (Header trailer : trailers) {
             try {
-                message += URLDecoder.decode(trailer.toString(), "UTF-8") + "\n";
+                messageBuilder.append(URLDecoder.decode(trailer.toString(), "UTF-8"));
             } catch (UnsupportedEncodingException e) {
-                message += trailer + "\n";
+                messageBuilder.append(trailer);
             }
+            messageBuilder.append('\n');
         }
-        return message;
+        return messageBuilder.toString();
     }
 
 }

--- a/src/main/java/com/amazonaws/services/neptune/rdf/io/NeptuneExportSparqlRepository.java
+++ b/src/main/java/com/amazonaws/services/neptune/rdf/io/NeptuneExportSparqlRepository.java
@@ -113,7 +113,7 @@ public class NeptuneExportSparqlRepository extends SPARQLRepository {
             chunkedInStream = (ChunkedInputStream) responseInStream;
         }
         else if (responseInStream instanceof EofSensorInputStream) {
-            // HTTPClient 4.5.13 provides no methods for accessing trailers from a wrapped stream requiring the use
+            // HTTPClient 4.5.13 provides no methods for accessing trailers from a wrapped stream requiring the use of
             // reflection to break encapsulation. This bug is being tracked in https://issues.apache.org/jira/browse/HTTPCLIENT-2263.
             try {
                 Method getWrappedStream = EofSensorInputStream.class.getDeclaredMethod("getWrappedStream");

